src/main/cpp/states/ballhopper/BallHopperStateMgr.cpp

```diff
@@ -0,0 +1,187 @@
+        if ( stateStringToEnumItr != stateMap.end() )
+        {
+            auto stateEnum = stateStringToEnumItr->second;
+            if ( m_stateVector[stateEnum]  == nullptr)
+            {
+                auto controlData = td->GetController();
+                auto target = td->GetTarget();
+                switch ( stateEnum)
+                {
+                    case BALL_HOPPER_STATE::OFF:
+                    {
+                        auto thisState = new BallHopperState( controlData, target );
+                        m_stateVector[stateEnum] = thisState;
+                        m_currentState = thisState;
+                        m_currentStateEnum = stateEnum;
+                        m_currentState->Init();
+                    }
+                    break;
+
+                    case BALL_HOPPER_STATE::HOLD:
+                    {
+                        auto thisState = new BallHopperState( controlData, target );
+                        m_stateVector[stateEnum] = thisState;
+                    }
+                    break;
+
+                    case BALL_HOPPER_STATE::SLOW_RELEASE:
+                    {
+                        auto thisState = new BallHopperSlowRelease( controlData, target, GetState(BALL_HOPPER_STATE::HOLD), GetState(BALL_HOPPER_STATE::RAPID_RELEASE) );
+                        m_stateVector[stateEnum] = thisState;
+                    }
+                    break;
+
+                    case BALL_HOPPER_STATE::RAPID_RELEASE:
+                    {
+                        auto thisState = new BallHopperState( controlData, target );
+                        m_stateVector[stateEnum] = thisState;
+                    }
+                    break;
+
+                    default:
+                    {
+                        Logger::GetLogger()->LogError( Logger::LOGGER_LEVEL::ERROR_ONCE, string("BallHopperStateMgr::BallHopperStateMgr"), string("unknown state"));
+
+                    }
+                    break;
+                }
+            }
+            else
+            {
+                Logger::GetLogger()->LogError( Logger::LOGGER_LEVEL::ERROR_ONCE, string("BallHopperStateMgr::BallHopperStateMgr"), string("multiple mechanism state info for state"));
+            }
+            
+        }
+        else
+        {
+            Logger::GetLogger()->LogError( Logger::LOGGER_LEVEL::ERROR_ONCE, string("BallHopperStateMgr::BallHopperStateMgr"), string("state not found"));
+        }
+    }
+}
+
+/// @brief run the current state 
+/// @return void
+void BallHopperStateMgr::RunCurrentState()
+{
+    if( MechanismFactory::GetMechanismFactory()->GetBallHopper().get() != nullptr)
+    {
+        if ( m_currentStateEnum == BALL_HOPPER_STATE::OFF && m_currentStateEnum != BALL_HOPPER_STATE::OFF)
+        {
+            SetCurrentState( BALL_HOPPER_STATE::OFF, false);
+        }
+        else if ( m_currentStateEnum == BALL_HOPPER_STATE::HOLD && m_currentStateEnum != BALL_HOPPER_STATE::HOLD)
```

Comment:
Delete; this shouldn't be this complex just run the current state (if there is no current state, log an error)